src/range_proof.rs

```diff
@@ -351,15 +519,71 @@ where
                     d1: ip_data.d1()?,
                     li: ip_data.li_compressed()?,
                     ri: ip_data.ri_compressed()?,
-                    extension_degree: statement.generators.extension_degree(),
+                    extension_degree,
                 });
             }
         }
     }
 
+    /// Create a single or aggregated range proof for a single party that knows all the secrets
+    /// The prover must ensure that the commitments and witness opening data are consistent
+    pub fn prove(
+        transcript_label: &'static str,
+        statement: &RangeStatement<P>,
+        witness: &RangeWitness,
+        seed_nonce_pair: &Option<RangeSeedNonce>,
+    ) -> Result<Self, ProofError> {
+        // Sanity check input data
+        if witness.extension_degree != statement.generators.extension_degree() {
+            return Err(ProofError::InvalidArgument(
+                "Witness and statement extension degrees do not match!".to_string(),
+            ));
+        }
+
+        // Extract the seed nonces for separation
+        let seed_nonce_helper = seed_nonce_pair.as_ref().map(|p| p.seed_nonce_helper.clone());
+        let seed_nonce_signer = seed_nonce_pair.as_ref().map(|p| p.seed_nonce_signer.clone());
+
+        // Extract values from the witness for use by the helper
+        let values = witness.openings.iter().map(|o| o.v).collect::<Vec<u64>>();
+
+        // Prepare the signer bootstrap data
+        let signer_bootstrap = SignerBootstrapData {
```

Comment:
Yeah, this will be too large for a ledger.

Reply:
See other comment about generators.